Framework/API/src/WorkspaceGroup.cpp

```diff
@@ -278,21 +279,36 @@ void WorkspaceGroup::workspaceDeleteHandle(
 }
 
 /**
- * Callback when a after-replace notification is received
- * Replaces a member if it was replaced in the ADS.
- * @param notice :: A pointer to a workspace after-replace notificiation object
+ * Callback when a before-replace notification is received
+ * Replaces a member if it was replaced in the ADS and checks
+ * for duplicate members within the group
+ * @param notice :: A pointer to a workspace before-replace notification object
  */
-void WorkspaceGroup::workspaceReplaceHandle(
+void WorkspaceGroup::workspaceBeforeReplaceHandle(
     Mantid::API::WorkspaceBeforeReplaceNotification_ptr notice) {
   std::lock_guard<std::recursive_mutex> _lock(m_mutex);
 
   const std::string replacedName = notice->objectName();
```

Comment:
Can we improve this method a bit more by removing its dependency on names completely? A replace is just saying that essentially the "old" object is about the be deleted and replaced by this new one. I don't see a reason why we should query the names at all.

If we just check for matching pointers on the `oldObject` and the contents and remove any matching pointer then wouldn't that take care of everything?